File: yh_star/src/yh_star_pub.cpp
#include "ros/ros.h" 
#include "yh_star/yh_star_msg.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_star_pub"); 
    ros::NodeHandle nh; 
 
    ros::Publisher pub = nh.advertise<yh_star::yh_star_msg>("yh_star_topic", 100);

    ros::Rate loop_rate(2);

    yh_star::yh_star_msg msg;
    int cnt = 0;

    while (ros::ok())
    {
        msg.stamp = ros::Time::now(); 
        msg.data = cnt; 
        ROS_INFO("send msg = %d", msg.stamp.sec); 
        ROS_INFO("send msg = %d", msg.stamp.nsec); 
        ROS_INFO("send msg = %d", msg.data); 
        cnt++;
        pub.publish(msg); 
        loop_rate.sleep(); 
    }

    return 0;
}

File: yh_star/scripts/yh_star_pub.py
#!/usr/bin/python3
# -*- coding: utf8 -*- 


import rospy
from yh_star.msg import yh_star_msg

def talker():
    pub = rospy.Publisher("yh_star_topic", yh_star_msg, queue_size = 100)
    rospy.init_node("yh_star_pub", anonymous=True)
    loop_rate = rospy.Rate(5)
    cnt = 0
    msg = yh_star_msg()

    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt
        rospy.loginfo("send msg = %d", msg.stamp.secs)
        rospy.loginfo("send msg = %d", msg.stamp.nsecs)
        rospy.loginfo("send msg = %d", msg.data)
        cnt += 1
        pub.publish(msg)
        loop_rate.sleep()

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass 